engine/src/main/java/org/hawkular/alerts/engine/impl/ispn/IspnAlertsServiceImpl.java

```diff
@@ -113,26 +114,28 @@ public void init() {
                     String op;
                     String value;
                     if (tokens.size() == 3) {
-                        query.append("'").append(tag).append("' and (");
                         op = tokens.get(1);
                         value = tokens.get(2);
                         boolean isRegexp = value.startsWith("'");
-                        String regexp = value.substring(1, value.length() - 1);
-                        regexp = regexp.equals("*") ? ".*" : regexp;
+                        String regexp = "";
+                        if (isRegexp) {
+                            regexp = value.substring(1, value.length() - 1);
+                            regexp = regexp.equals("*") ? ".*" : regexp;
+                        }
                         if (op.equalsIgnoreCase(EQ)) {
                             // tag =
                             if (isRegexp) {
-                                query.append("/").append(regexp).append("/");
+                                query.append("/").append(tag).append(TagsBridge.SEPARATOR).append(regexp).append("/");
```

Comment:
The new format how internally tags are represented in the index engine is defined in TagsBridge class.
Potential constants to build expression will be pointed from there.